Challenge/javascript-onboarding/src/problem2.js

```diff
@@ -1,6 +1,49 @@
+function isDuplicate(string, index) {
+  return (
+    string[index - 1] === string[index] || string[index] === string[index + 1]
+  );
+}
+
+function deleteDuplicateCharacter(string) {
+  return Array.from(string)
+    .filter((_, index) => !isDuplicate(string, index))
+    .join("");
+}
+
+function deleteAllDuplicateCharacter(string) {
+  const result = deleteDuplicateCharacter(string);
+
+  return string.length !== result.length
+    ? deleteAllDuplicateCharacter(result)
+    : result;
+}
+
+const ErrorMessage = Object.freeze({
+  ERROR_INVALIDE_CRYPTOGRAM: "입력된 암호문의 형태가 유효하지 않습니다.",
+  ERROR_INVALIDE_CRYPTOGRAM_LENGTH: "입력된 암호문의 길이가 올바르지 않습니다.",
+});
+
+function validateCryptogram(cryptogram) {
+  if (cryptogram.toLowerCase() !== cryptogram)
+    throw new Error(ErrorMessage.ERROR_INVALIDE_CRYPTOGRAM);
+  if (cryptogram.length < 0 || cryptogram.length > 1000)
+    throw new Error(ErrorMessage.ERROR_INVALIDE_CRYPTOGRAM_LENGTH);
+}
+
```

Comment:
여기서 "a1" 같은 문자열이 나오면 예외가 발생해야 하지 않을까요?

Reply:
아 맞아요 나중에 처리한다는 것을 깜빡을..

정규표현식 사용해서 알파벳 소문자만 입력될 수 있게 변경했습니다~!

```suggestion
function validateCryptogram(cryptogram) {
  if (cryptogram.match(/[a-z]/g).join("") !== cryptogram)
    throw new Error(ErrorMessage.ERROR_INVALIDE_CRYPTOGRAM);
  if (cryptogram.length < 0 || cryptogram.length > 1000)
    throw new Error(ErrorMessage.ERROR_INVALIDE_CRYPTOGRAM_LENGTH);
}
```